sw/device/silicon_creator/lib/ownership/ownership.c

```diff
@@ -0,0 +1,102 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/ownership/ownership.h"
+
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/silicon_creator/lib/drivers/flash_ctrl.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/ownership/ecdsa.h"
+#include "sw/device/silicon_creator/lib/ownership/owner_keys.h"
+#include "sw/device/silicon_creator/lib/ownership/ownership.h"
+
+// RAM copy of the owner INFO pages from flash.
+owner_block_t owner_page[2];
+
+rom_error_t ownership_init(void) {
+  flash_ctrl_perms_t perm = {
+      .read = kMultiBitBool4True,
+      .write = kMultiBitBool4True,
+      .erase = kMultiBitBool4True,
+  };
+  flash_ctrl_info_perms_set(&kFlashCtrlInfoPageOwnerSlot0, perm);
+  flash_ctrl_info_perms_set(&kFlashCtrlInfoPageOwnerSlot1, perm);
```

Comment:
will the owner pages also have ecc and scrambling set on them?

Reply:
Fixed.